src/Microsoft.DotNet.Interactive.Http.Parsing/Parsing/HttpVariableDeclarationNode.cs

```diff
@@ -17,7 +17,7 @@ internal HttpVariableDeclarationNode(SourceText sourceText, HttpSyntaxTree? synt
     {
     }
 
-    public string VariableName => ChildTokens.Where(t => t.Kind == HttpTokenKind.Word).Select(t => t.Text).FirstOrDefault() ?? string.Empty;
+    public string VariableName => string.Join("", ChildTokens.Where(t => t.Kind == HttpTokenKind.Word || (t.Kind == HttpTokenKind.Punctuation && t.Text is "_")).Select(t => t.Text)) ?? string.Empty;
```

Comment:
Should we report an error if a variable name contains characters like `-` that it should not contain?

Reply:
This suggests an alternative approach, which is to parse everything but whitespace into the variable declaration node, and then report diagnostics for invalid characters.

Reply:
Note that characters like `-` will be allowed on the RHS of the variable declaration (if they appear inside a string literal or embedded expression for example) which is why I think having dedicated parsing for the name portion may be helpful.

Reply:
Ah never mind, I was confused by the naming. I assumed the HttpVariableDeclarationNode encompasses the entire declaration (inlcuding the = and the expression on the RHS). Looks like that is the HttpVariableDeclarationAndAssignmentNode :) So please ignore my last comment above.

Reply:
(I also wonder if the naming could be made less confusing - but perhaps its ok since it was not as confusing once I looked at the names of all the involved node types :))